### DISCUSSION TITLE
Is it possible to know with which version of AIR was aab/apk built?

### DISCUSSION THREAD

**Discussion by GeneralVimes** [airsdk/Adobe-Runtime-Support#3170] [Q&A] [answered]
That's the question :) I noticed that in the past, about 10 months ago, there was a time when ANRs from my game was quite low, but after one of the updates it began to grow. I'd like to rebuild a game with an older version of AIR and resubmit it, but I don't remember which AIR version I used then. Is it possible to know this somehow, by opening aab or apk?

**Comment by ajwfrost** [comment, marked as answer]
There may be a few ways of doing this... the one I just tried was to pull out the "libCore.so" from the APK file (under a lib folder) and then grep it:
```
$ grep -ao '[0-9][0-9]\.[0-9]\.[0-9]\.[0-9]' libCore.so
```
which should display the version string (as well as most of the 'localhost' IP address a few times!)

There's a class definition in the dex stuff but I'm not sure how easy that would be to get at... but in case it helps:
```
air.com.adobe.appentry.GetVersionCode.AIR_BUILD_VERSION
```

And then if you were able to change the SWF file in your application (unzip the apk, change it, re-zip and re-sign), you could try checking the AS3 Capabilities version value...

But also ... I think just run it up with ADB connected. Fairly sure AIR always prints out the version and CPU architecture as it starts up....
> 10-30 08:47:43.706 12323 12323 I AIR VERSION: :50.2.3.6
> 10-30 08:47:43.706 12323 12323 I DEVICE ARCH: :arm64-v8a

**Reply by GeneralVimes** [reply]
Thank you so much!

**Reply by itlancer** [reply]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/10899066/4907d8fa-9923-4d40-adf2-b930b364f121)
Also you can see it in Scout.

**Reply by MalacTheLittle** [reply]
Is there a way to get AIR ARCH dynamically?
I need to know if app is built for armv7 or armv8...

thanks

**Reply by ajwfrost** [reply]
You can use `flash.system.Capabilities` - the `cpuArchitecture` for the family and `cpuAddressSize` for the 32 vs 64 bit.
https://airsdk.dev/reference/actionscript/3.0/flash/system/Capabilities.html
